src/ompl/geometric/planners/rrt/RRTstar.h

```diff
@@ -148,42 +160,125 @@ namespace ompl
                 return delayCC_;
             }
 
-            /** \brief Controls whether the tree is pruned during the search. */
-            void setPrune(const bool prune)
+            /** \brief Controls whether the tree is pruned during the search. This pruning removes
+                a vertex \e and \e its \e descendants if the lower-bounding estimate of a solution
+                constrained to pass the the \e vertex  is greater than the current solution.
+                As the ancestory of the descendent vertices can change at anytime, this means that the removed
+                descendents may actually be capable of later providing a better solution once
+                their incoming path passes through a different vertex (e.g., a change in homotopy class) */
+            void setTreePruning(const bool prune)
             {
-                prune_ = prune;
+                useTreePruning_ = prune;
             }
 
             /** \brief Get the state of the pruning option. */
-            bool getPrune() const
+            bool getTreePruning() const
             {
-                return prune_;
+                return useTreePruning_;
             }
 
-            /** \brief Set the percentage threshold (between 0 and 1) for pruning the tree. If the new tree has removed
-                at least this percentage of states, the tree will be finally pruned. */
-            void setPruneStatesImprovementThreshold(const double pp)
+            /** \brief Set the fractional change in solution cost necessary for pruning to occur, i.e.,
+                prune if the new solution is at least X% better than the old solution.
+                (e.g., 0.0 will prune after every new solution, while 1.0 will never prune.) */
+            void setPruneThreshold(const double pp)
             {
-                pruneStatesThreshold_ = pp;
+                pruneThreshold_ = pp;
             }
 
             /** \brief Get the current prune states percentage threshold parameter. */
-            double getPruneStatesImprovementThreshold () const
+            double getPruneThreshold() const
+            {
+                return pruneThreshold_;
+            }
+
+            /** \brief Controls whether heuristic rejection is used on samples (e.g., x_rand) */
+            void setSampleRejection(const bool reject);
```

Comment:
Sorry, that statement was a bit vague. I meant there's no (long-term) need to support both rejection sampling and informed sampling of an admissible heuristic, as

1. Informed sampling is better than rejection sampling (working on a paper to show this), and
2. The way we wrote the classes, an objective without informed sampling just defaults to rejection sampling.

For now, I do think this should remain an option though in case someone wants to reject on a inadmissible heuristic.